packages/core/src/components/section/section.tsx

```diff
@@ -34,6 +34,31 @@ import { Icon } from "../icon/icon";
  */
 export type SectionElevation = typeof Elevation.ZERO | typeof Elevation.ONE;
 
+export interface SectionCollapseProps
+    extends Pick<CollapseProps, "className" | "isOpen" | "keepChildrenMounted" | "transitionDuration"> {
+    /**
+     * Whether the component is initially open or closed.
+     *
+     * This prop has no effect if `collapsible={false}` or the component is in controlled mode,
+     * i.e. when `isOpen` is **not** `undefined`.
+     *
+     * @default true
+     */
+    defaultIsOpen?: boolean;
+
+    /**
+     * Whether the component is open or closed.
+     *
+     * Passing a boolean value to `isOpen` will enabled controlled mode for the component.
+     */
+    isOpen?: boolean;
+
+    /**
+     * Callback invoked in controlled mode when the collapse element is clicked.
+     */
+    onToggleCollapse?: () => void;
```

Comment:
minor: I think we can abbreviate this to simply `onToggle`, as it's already inside a `collapseProps` key in `SectionProps`

```suggestion
    onToggle?: () => void;
```